From 950e54eb0887eda313a905af6b40ad01854a95bd Mon Sep 17 00:00:00 2001
From: Eirik Aavitsland <eirik.aavitsland@qt.io>
Date: Thu, 29 Sep 2022 14:00:32 +0200
Subject: [PATCH 09/12] Fix bundled libtiff compilation lzw warning and reading
 failure

A lacking #include of stdlib.h leads to a compilation warning and
runtime reading failure on some platforms. Observed on MSVC2019 in
debug mode.

Fix by applying commit cbd71793f58c0eda8128f799d415c56e5c64eaa5 in the
upstream libtiff repo.

Fixes: QTBUG-107040
Pick-to: 6.4 6.2 5.15
Change-Id: Ic76c4627632302423d0e0f8f1b51df342e62b420
Reviewed-by: Paul Olav Tvete <paul.tvete@qt.io>
(cherry picked from commit 64e455da1c05a9fafe7125690053fb951268a733)
---
 src/3rdparty/libtiff/libtiff/tif_lzw.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/3rdparty/libtiff/libtiff/tif_lzw.c b/src/3rdparty/libtiff/libtiff/tif_lzw.c
index 096824d..8c3205f 100644
--- a/src/3rdparty/libtiff/libtiff/tif_lzw.c
+++ b/src/3rdparty/libtiff/libtiff/tif_lzw.c
@@ -39,6 +39,7 @@
 
 #include <stdbool.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 /* Select the plausible largest natural integer type for the architecture */
 #define SIZEOF_WORDTYPE SIZEOF_SIZE_T
-- 
2.42.0

